main.tf

```diff
@@ -52,3 +45,17 @@ resource "snowflake_user" "this" {
   rsa_public_key   = local.rsa_public_key
   rsa_public_key_2 = var.rsa_public_key_2
 }
+
+resource "snowflake_role_grants" "default_role" {
+  count = var.grant_default_roles && var.default_role != null ? 1 : 0
+
+  role_name = var.default_role
+  users     = [one(resource.snowflake_user.this[*].name)]
+}
+
+resource "snowflake_role_grants" "default_secondary_roles" {
+  for_each = var.grant_default_roles ? toset(var.default_secondary_roles) : []
```

Comment:
same here: check for `module.this.enabled`? (IIUC that resource will be created even though an entire module is disabled, right?)

Reply:
Good catch - added